### DISCUSSION TITLE
Porting WiFi Manager to RTL8720?

### DISCUSSION THREAD

**Discussion by S10143806H** [khoih-prog/WiFiWebServer_RTL8720#7] [Q&A] [answered]
Hi @khoih-prog.
I am from [Ameba IoT [RTL8722/RTL8195/RTL8710/...]](https://www.facebook.com/groups/AmebaIoT/) FB group with 4.4k active users. Recently, there are some requirements in the group requesting for porting WiFi Manager from ESP to Ameba. And I have just noticed that you have already contributed to another similar repo: [ESP_WifiManager](https://github.com/khoih-prog/ESP_WiFiManager). 
**May I know if there are any chances that WiFi Manager will be ported in this repo as well in the future?**

We would like to promote your repo in our group as well if you don't mind! 
Thank you.

**Comment by khoih-prog** [comment, marked as answer]
HI @S10143806H 

Thanks for your and Ameba's interest in the library.

The infrastructure for the future so-called **RTL8720_WiFiManager** has been ready thanks for these libraries of mine

1. [**FlashStorage_RTL8720**](https://github.com/khoih-prog/FlashStorage_RTL8720)
2. [**DoubleResetDetector_Generic**](https://github.com/khoih-prog/DoubleResetDetector_Generic)
3. This [**WiFiWebServer_RTL8720**](https://github.com/khoih-prog/WiFiWebServer_RTL8720)

I can either add the support of RTL8720 into [**WiFiManager_Generic_Lite** library](https://github.com/khoih-prog/WiFiManager_Generic_Lite) or create the new **RTL8720_WiFiManager** and will decide when I have time to start that new work.

I dropped the development of RTL8720-related libraries because there was little interest / users at that time. I'm hoping that the market will understand and value the superior 5GHz-band WiFi support of RTP8720 boards.

You can also start the development, if interested and urgent. Just follow the [**WiFiManager_Generic_Lite** library](https://github.com/khoih-prog/WiFiManager_Generic_Lite), then make a PR or create the new **RTL8720_WiFiManager**

**Reply by S10143806H** [reply]
Dear @khoih-prog,

Thank you for your prompt reply and for providing the [WiFiManager_Generic_Lite library](https://github.com/khoih-prog/WiFiManager_Generic_Lite). I will take a look at I and get back to you. 

I will mark this discussion as *answered* first. Please feel free to join Ameba's community. Meanwhile, you could check our porting for Arduino on RTL8722DM, RTL8720DN ... here [AmebaD Arduino](https://github.com/ambiot/ambd_arduino).

Thanks again!

**Comment by khoih-prog** [comment]
I've just checked and turned out that I have written a version of `WiFiManager_RT8720` sometime ago, but not published yet.

It still needs some more tests to finalize.

@S10143806H 

Are you and/or your friends can spend time to test and finalize that library and save us some time?
If OK, I can post / publish the library within today.

**Comment by khoih-prog** [comment]
I've just published the new [**WiFiManager_RTL8720** library](https://github.com/khoih-prog/WiFiManager_RTL8720)

Try the example [**RTL8720_WiFi**](https://github.com/khoih-prog/WiFiManager_RTL8720/tree/main/examples/RTL8720_WiFi)

There is a severe bug in WiFi library of AmebaD core. Please check [**Important Notes**](https://github.com/khoih-prog/WiFiManager_RTL8720#important-notes) then push Realtek to fix.

**Reply by S10143806H** [reply]
Hi @khoih-prog ,

Sorry for the late reply since I just back from holiday. I have tested from my side that the basic example: [RTL8720_WiFi](https://github.com/khoih-prog/WiFiManager_RTL8720/tree/main/examples/RTL8720_WiFi) is working well. 

![WifiManager_Test](https://user-images.githubusercontent.com/9852678/172088408-66c0d559-09cb-4c9b-95ab-14833e34ffe6.png)

Moving on to reproduce the issue of the `WiFi.status()` bug now. 
I will be moving on the discussion to the [WiFiManager_RTL8720 library](https://github.com/khoih-prog/WiFiManager_RTL8720) repo.

Thank you.
